sycl/doc/design/CompileTimeProperties.md

```diff
@@ -497,31 +520,14 @@ define void @foo(i32* %ptr) {
     i8* getelementptr inbounds ([16 x i8], [16 x i8]* @.str, i64 0, i64 0),
     i8* getelementptr inbounds ([9 x i8], [9 x i8]* @.str.1, i64 0, i64 0),
     i32 3,
-    i8* bitcast ({ [9 x i8]*, i8*, [9 x i8]*, i32 }* @.args to i8*))
+    i8* bitcast ({ [9 x i8]*, i8*, [9 x i8]*, [3 x i8]* }* @.args to i8*))
 
   %3 = bitcast i8* %2 to i32**
   store i32* %ptr, i32** %3
   ret void
 }
 ```
 
-The front-end encodes the properties from the C++ attribute
-`[[__sycl_detail__::add_ir_annotations_member()]]` into the
-`@llvm.ptr.annotation` call as follows:
-
-* The first parameter to `@llvm.ptr.annotation` is the pointer to annotate (as
-  with any call to this intrinsic).
-* The second parameter is the literal string `"sycl-properties"`.
-* The third parameter is the name of the source file (as with any call to this
-  intrinsic).
-* The fourth parameter is the line number (as with any call to this intrinsic).
-* The fifth parameter is a metadata tuple with information about all of the
-  properties.  The first element of the tuple is a string literal with the name
-  of the first property.  The second element is the value of the first
-  property.  The third element is a string literal with the name of the second
-  property, etc.  Since each property has exactly one value, this tuple has an
-  even number of elements.
-
 **NOTE**: Calls to the `@llvm.ptr.annotation` intrinsic function are known to
 disable many clang optimizations.  As a result, properties added to a
 non-global variable will likely result in LLVM IR (and SPIR-V) that is not well
```

Comment:
GitHub won't let me add a comment to the lines below, but I think the section "IR representation via @llvm.ptr.annotation:" needs some updates.  In particular, this sentence is no loner true:

> Since metadata types are not limited to strings, there is no need to convert
> the property values to strings.

Reply:
Thank you for pointing it out! I completely missed that section. I have reworded it.